pkg/docker/config/config.go

```diff
@@ -50,10 +50,17 @@ var (
 	ErrNotLoggedIn = errors.New("not logged in")
 	// ErrNotSupported is returned for unsupported methods
 	ErrNotSupported = errors.New("not supported")
+	// ErrNamespacedRegistry is returned if namespaced registries are not
+	// supported for the credential helper. Only the
+	// sysregistriesv2.AuthenticationFileHelper supports namespaced registries.
+	ErrNamespacedRegistry = errors.New("namespaced registry not supported for credential helper")
 )
 
 // SetCredentials stores the username and password in the credential helper or file
 // and returns path to file or helper name in format (helper:%s).
+// An ErrNamespacedRegistry will be part of the returned multierror if the
+// provided registry is namespaced and a credentials helper other than the
+// AuthenticationFileHelper is used.
```

Comment:
Hum… do we want this as a multi-error component or as a single top-level error?
I.e. the caller gets one of:
- success
- error indicating a failure to write to file, _and nothing else_ (even if there are helpers; or should there be one `ErrNamespacedRegistry` in that case?)
- `ErrNamespacedRegistry` only (if files aren't used, and even if there are many helpers)

_What does c/common/pkg/auth need for a reasonable error message?_

3 repetitions of `ErrNamespacedRegistry` with nothing else are not really useful; at that point 3 untyped error messages to the effect of “helper $foo is configured, but we are not storing namespace credentials there” would work at least as well.

I imagine that c/common/pkg/auth would like a single top-level `ErrNamespacedRegistry` to check against, if that’s the _only_ reason writing fails, so that it can write a nice error message with explanations. Or maybe c/image should just return such a nice error message already, with pkg/auth not having to care, and c/image doesn’t need to define an API-visible error.

Reply:
Hm, we can wrap the error with the name of the credentials helper to provide more context. I decided to implement that, not sure if it's ideal.

Reply:
Works for me.

If the caller is not going to be specifically checking for this value (and c/common currently isn’t), a `fmt.Errorf("whatever_we_decide_to_call_namespaced_registry is not supported for credential helper %s", helperName)` is easier to understand than wrapping, and we don’t need to expose an API.

Reply:
SGTM, improved the wrapping as suggested